### ISSUE TITLE
❤︎ ⇢ ❤️ PLZ CONTRIBUTE TO THIS maintained FORK ▸ https://github.com/mattr-/slate ❤️ ⇠ ❤︎

### ISSUE THREAD

**Issue by MoOx** [jigish/slate#437]
_Edit_

According to the non response of jigish, here is a fork that seems a better choice
# [mattr-/slate](https://github.com/mattr-/slate)

---

Hi @jigish,

First of all thanks for your awesome work. I thank you in my head each time I'm triggering my layout adjustment command :)
I've tried every window manager I've found for OS X & the one you did is still the best we can find these days. JavaScript configuration file is just the icing on the cake !

According to your GitHub activity I understand that you are pretty busy with other stuff in your life than this application (& that's a good thing for you). However I saw over there so many issues & pull requests that I'm wondering about the evolution of this app. I don't like to see apps I like & use to start going down because the maintainer is a busy man (I am a busy man too, we all are & that's good for our lifes).

So what about to cherry pick some people to give them write access in this repos? It can be a good thing to keep the project up to date with the OS & community needs.
I know this idea seems crazy, but I really think this can be a good thing for your project. You can keep an eye over there & give your opinion (I am sure you still are interested by the project).
You can also just say "yes please do that" or "no I don't want that for this app". You can still be the leader of course.

Please think about this idea :)

**Comment by SebastienElet** [comment]
:+1:

**Comment by mattr-** [comment]
Opening this app up to the community is something that's already sort of been done. I've forked this repo to mattr-/slate, have done one release, and am in the process of getting things set up and streamlined so I can do more releases (including signing them so that they're easier for folks to install). Once these things are set up, I intend to make a larger announcement.

**Comment by ghost** [comment]
I'd like to be a contributor if possible.

@mattr- the advantage of keeping the repo here is that there are so many people following it and it has many issues already submitted. Hopefully @jigish will allow several of us write access to the repo so your fork can be merged back here.

**Comment by mattr-** [comment]
Having access to this repo would be ideal, but the emails I've sent to @jigish have gone unreplied to, so I started my own fork. :sob:

**Comment by ghost** [comment]
That's a bummer... I'm sure lots of people feel the same way. Maybe there is a chance to rebrand this project and start a new repo?

**Comment by MoOx** [comment]
Oh if he is ignoring your message, let's discuss about a fork + rebrand indeed.
What about Slate2.app ? :D
Maybe you have better ideas :)
Can't come up with a good name when I think about "window" and "management" :D

**Comment by mattr-** [comment]
I've sent another email and am hoping for a response.

**Comment by sdegutis** [comment]
He's not going to respond. Many people have contacted him over the past two years. He's [clearly active on Github](https://github.com/jigish?tab=activity) but apparently wants nothing to do with Slate anymore. That's fine, that's his choice. But he's not going to respond. It'd be bette to just make a new issue with a very attention-grabbing title so that new people who come on here saying this same thing will see it and read it and know where to go from there instead of repeating this.

**Comment by MoOx** [comment]
I can just rename this issue with a NAME LIKE THAT :)

**Comment by MoOx** [comment]
That's a shame @jigish just ignore all messages. Not a cool way to behave :(

**Comment by sdegutis** [comment]
@MoOx Meh. For all we know, he's had a terrible tragedy or something which put Github projects into perspective of being relatively unimportant. Who knows. We don't know the whole story, maybe he has a great reason. All I know is that it is what it is, and it's not likely to change.

**Comment by MoOx** [comment]
If he had times to comment issues or work on some open source dotfiles & vim-config, he could take 5 min to give some write access to some people to make a second life to this project. :/
But I understand your argument.

**Comment by MoOx** [comment]
Still no update. Time to rename this issue.
Who had a interesting fork?

**Comment by troxler** [comment]
The one of @mattr- (https://github.com/mattr-/slate) seems to be the most active one according to https://github.com/jigish/slate/network.

**Comment by MoOx** [comment]
I just updated this issue with an attractive name. What do you think about that guys?

**Comment by sdegutis** [comment]
@MoOx The unicode heart isn't visible on Github, only in my browser's (Chrome's) tab.

**Comment by MoOx** [comment]
You need to use a real browser :trollface: 

![screen shot 2015-01-14 at 16 33 23](https://cloud.githubusercontent.com/assets/157534/5741399/1e755b92-9c0b-11e4-81f9-dc25b21178d2.png)

I just updated the title. Is it better this way?

**Comment by sdegutis** [comment]
Yep.

**Comment by jiMuBao** [comment]
https://github.com/ianyh/Amethyst very active windows tiling manager

**Comment by jigish** [comment]
hey @mattr- @MoOx @asethwright  --

First of all, my apologies for being _very_ absent over the past couple of years. I've had some serious work-life balance issues and only been contributing to projects I've used for work. That has changed, though.

It has been a while since I first wrote Slate and am planning on pushing a complete rewrite to my github (probably with in ~2 weeks). As it is a complete rewrite, Slate2 will not support many features up front and the config has changed quite a bit. In addition I'm thinking about supporting only Yosemite+ with Slate2. That being said, I will be committing to it very actively to bring it up to speed with Slate.

As far as this version of Slate is concerned I'm happy to do any of the following:
1. add a pointer to @mattr-'s fork to the readme
2. merge a large pull request and give write access to @mattr- 

either works for me, but I will be putting a pointer to Slate2 on the readme once I push it to my github.

again, apologies for my absence.

**Comment by jigish** [comment]
Another option would be to commit the rewrite into this repository and add a pointer in the readme to @mattr-'s fork for anyone who wants the original slate. I feel like as it was my first foray into OS X development I've learned a _lot_ from it and would like to do a complete rewrite from scratch.

**Comment by sdegutis** [comment]
@jigish
- +1 for the Yosemite+ requirement.
- Is your upcoming rewrite in Swift?
- Are you going to use the `JSContext` class instead of a hidden WebView?
- You might wanna ask for some volunteer contributors who can help clean up the Github Issues that have accumulated and maybe help with the readme.

**Comment by sdegutis** [comment]
@jigish Also, feel free to use https://github.com/sdegutis/WindowManager as a baseline for your rewrite if you choose Swift.

**Comment by jigish** [comment]
yes to Swift and `JSContext`. I actually have a lot of the basics written already, just need to implement to more complex stuff. I'll definitely take a look at WindowManager -- thanks!

If we decide to maintain the original Slate here then I'd suggest just wiping all github issues and pull requests with a generic "I'm wiping this please reopen if it is still an issue" message.

**Comment by sdegutis** [comment]
@jigish Also, one approach that might be worth considering is to use Lua instead of JS. Lua is very similar to JS semantically, but much easier to integrate with than JSContext. Plus, Apple's rules around converting structures to/from JS are not super well defined, and there are times it will do a different thing than expected. Plus, the ObjC API for it does not integrate with Swift classes, you have to write NSObject-based classes with them. Since Lua has a C API, I wrote a Swift wrapper as https://github.com/sdegutis/lua4swift which you can use if you want. Just throwing it out there as a possibility.

**Comment by mattr-** [comment]
@jigish I'm happy to continue maintaining this version of Slate, so however you want to handle the rewrite, whether in this repo or in a separate repo, it doesn't really matter to me. :smiley:

**Comment by mattr-** [comment]
@jigish If you do end up using the separate repo, I think it makes sense for me to have access to this repo. Another option is to create an organization and move this repo and the new repo for Slate2 into repos under the organization. I know GitHub support is generally happy to do those sorts of things for us.

**Comment by rjhilgefort** [comment]
@sdegutis If you (or anyone) is wanting to go the Lua route, I would recommend just checking out [Hammerspoon](http://www.hammerspoon.org/) and building from there. I've actually already started a pseudo slate rewrite (similar config principles) and would be happy to work with you in going that route. Lmk if your interested and I can push up my project to GitHub.

:+1: @jigish for resurfacing. Looking forward to the rewrite.

**Comment by sdegutis** [comment]
@rjhilgefort Thanks for the suggestion will look into it.

**Comment by nudefireninja** [comment]
Finally, the mastermind responds! And a new version of Slate is in the pipes? Awesome!

However, I have an ongoing interest in the current version of Slate because I'm one of those people who still use OS X Snow Leopard. I use both 10.6.8 and 10.9.5 (soon 10.10.3) depending on what software I need to use, so I look forward to installing Slate2 on my Yosemite partition, but I'm going to continue using Slate on my preferred version of OS X.

With that said, could someone (@mattr-?) _pleeeease_ implement this: https://github.com/jigish/slate/issues/382 It's my biggest wish for Slate.

**Comment by cmsj** [comment]
@rjhilgefort Thanks for the Hammerspoon plug, but @sdegutis knows about it. It's a fork of his awesome Mjolnir project :)

**Comment by chdiza** [comment]
So where's this new Slate2 we were teased with? :smile:

**Comment by arturhoo** [comment]
hey @jigish 

I have no issues or feature/improvement suggestions for Slate, and that has been the case for the past 3 years or so. Just wanted to to thank you for this amazing work that has become such an important part of my workflow.

Nice to hear that you've managed to find a healthy life/work balance and are willing to return to the oss community. Take as much time as you need ;)

**Comment by clemens-tolboom** [comment]
@sdegutis the link to https://github.com/sdegutis/WindowManager seems broken. Is it moved?

@jigish were izz the goodies ;-) aka slate2 located?

**Comment by clemens-tolboom** [comment]
@sdegutis guess now you meant https://github.com/sdegutis/AppGrid

**Comment by clemens-tolboom** [comment]
Via https://github.com/sdegutis/mjolnir I ended up choosing https://github.com/Hammerspoon/hammerspoon as my replacement for slate.

@jigish thanks for the past years
